numba/targets/arraymath.py

```diff
@@ -3948,3 +3948,89 @@ def np_kaiser_impl(M, beta):
         return _i0n(n, alpha, beta)
 
     return np_kaiser_impl
+
+
+@register_jitable
+def cross_operation(a, b, cp):
+    dta = a.dtype
+    a0 = a[..., 0]
+    a1 = a[..., 1]
+    if a.shape[-1] == 3:
+        a2 = a[..., 2]
+    else:
+        a2 = np.multiply(dta.type(0), a0)
+    dtb = b.dtype
+    b0 = b[..., 0]
+    b1 = b[..., 1]
+    if b.shape[-1] == 3:
+        b2 = b[..., 2]
+    else:
+        b2 = np.multiply(dtb.type(0), b0)
+
+    cp0 = np.multiply(a1, b2) - np.multiply(a2, b1)
+    cp1 = np.multiply(a2, b0) - np.multiply(a0, b2)
+    cp2 = np.multiply(a0, b1) - np.multiply(a1, b0)
+
+    cp[..., 0] = cp0
+    cp[..., 1] = cp1
+    cp[..., 2] = cp2
+
+
+@overload(np.cross)
+def np_cross(a, b):
+    if isinstance(a, types.Array) and isinstance(b, types.Array):
```

Comment:
Good point. I'll look into it.

Reply:
@stuartarchibald I'll wait for an example of the code gen "trick" you mentioned, since allowing array-likes in general would break the conditional statment where I check `ndims` to deliver one implementation or another (tuple doesn't have `ndims` atrr.).